### ISSUE TITLE
Factory::create() in data providers not working when using paratest

### ISSUE THREAD

**Issue by esnard** [zenstruck/foundry#733]
Hello,

I just upgraded to PHPUnit 11.4 and wanted to use `Factory::create()` in data providers.

Here's an example test I'm using:

```php
<?php

declare(strict_types=1);

namespace App\Tests;

use App\Fixtures\Factory\Main\PersonFactory;
use PHPUnit\Framework\Attributes\DataProvider;
use Zenstruck\Foundry\Test\Factories;

class TestTest extends KernelTestCase
{
    use Factories;

    #[DataProvider('providerTest')]
    public function testTest(): void
    {
        self::assertTrue(true);
    }

    public static function providerTest(): iterable
    {
        yield [PersonFactory::createOne()];
    }
}
```

Here's an extract from my PHPUnit configuration file:

```xml
    <extensions>
        <bootstrap class="Zenstruck\Foundry\PHPUnit\FoundryExtension" />
        <bootstrap class="DAMA\DoctrineTestBundle\PHPUnit\PHPUnitExtension" />
    </extensions>
```

PHPUnit output:

```
PHPUnit 11.4.4 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.3.13
Configuration: /app/phpunit.xml.dist

.                                                                   1 / 1 (100%)

Time: 00:00.287, Memory: 100.46 MB
```

ParaTest output:

```
ParaTest v7.6.0 upon PHPUnit 11.4.4 by Sebastian Bergmann and contributors.


In Configuration.php line 85:
                                                                                                                                    
  Foundry is not yet booted. Ensure ZenstruckFoundryBundle is enabled. If in a test, ensure your TestCase has the Factories trait.  
                                                                                                                                    

paratest [--functional] [-m|--max-batch-size MAX-BATCH-SIZE] [--no-test-tokens] [--passthru-php PASSTHRU-PHP] [-p|--processes PROCESSES] [--runner RUNNER] [--tmp-dir TMP-DIR] [-v|--verbose] [--bootstrap BOOTSTRAP] [-c|--configuration CONFIGURATION] [--no-configuration] [--cache-directory CACHE-DIRECTORY] [--testsuite TESTSUITE] [--exclude-testsuite EXCLUDE-TESTSUITE] [--group GROUP] [--exclude-group EXCLUDE-GROUP] [--filter FILTER] [--process-isolation] [--strict-coverage] [--strict-global-state] [--disallow-test-output] [--dont-report-useless-tests] [--stop-on-defect] [--stop-on-error] [--stop-on-failure] [--stop-on-warning] [--stop-on-risky] [--stop-on-skipped] [--stop-on-incomplete] [--fail-on-incomplete] [--fail-on-risky] [--fail-on-skipped] [--fail-on-warning] [--order-by ORDER-BY] [--random-order-seed RANDOM-ORDER-SEED] [--colors [COLORS]] [--no-progress] [--display-incomplete] [--display-skipped] [--display-deprecations] [--display-errors] [--display-notices] [--display-warnings] [--teamcity] [--testdox] [--log-junit LOG-JUNIT] [--log-teamcity LOG-TEAMCITY] [--coverage-clover COVERAGE-CLOVER] [--coverage-cobertura COVERAGE-COBERTURA] [--coverage-crap4j COVERAGE-CRAP4J] [--coverage-html COVERAGE-HTML] [--coverage-php COVERAGE-PHP] [--coverage-text [COVERAGE-TEXT]] [--coverage-xml COVERAGE-XML] [--coverage-filter COVERAGE-FILTER] [--no-coverage] [--] [<path>]
```

Am I missing something in my configuration, or are factories not available in data providers when using ParaTest yet?

**Comment by nikophil** [comment]
hum... paratest was just not considered when this feature was created :shrug: 

so... yes, it's not that strange that it does fail :grin: 

I really don't know how paratest handles data providers, so the bug might be tricky (this whole feature was actually tricky: it  required some modification in how phpunit is bootstraped)

**Comment by nikophil** [comment]
I'm not 100% sure, but I think paratest sould adapt its code toward some fixes in PHPUnit 11.4.

I've opened an issue there: https://github.com/paratestphp/paratest/issues/916

**Comment by nikophil** [comment]
@esnard it is now fixed in paratest 7.6.2
